includes/embeds/class-amp-youtube-embed-handler.php

```diff
@@ -83,69 +141,72 @@ public function shortcode( $attr ) {
 
 		$video_id = $this->get_video_id_from_url( $url );
 
-		return $this->render(
-			[
-				'url'      => $url,
-				'video_id' => $video_id,
-			]
-		);
+		return $this->render( compact( 'video_id' ), $url );
 	}
 
 	/**
 	 * Render oEmbed.
 	 *
 	 * @see \WP_Embed::shortcode()
+	 * @deprecated This is no longer being used.
 	 *
 	 * @param array  $matches URL pattern matches.
 	 * @param array  $attr    Shortcode attribues.
 	 * @param string $url     URL.
 	 * @return string Rendered oEmbed.
 	 */
 	public function oembed( $matches, $attr, $url ) {
+		_deprecated_function( __METHOD__, '1.3.1' );
 		return $this->shortcode( [ $url ] );
 	}
 
 	/**
-	 * Render.
+	 * Render embed.
 	 *
-	 * @param array $args Args.
+	 * @param array  $args Args.
+	 * @param string $url  URL.
 	 * @return string Rendered.
 	 */
-	public function render( $args ) {
+	public function render( $args, $url ) {
 		$args = wp_parse_args(
 			$args,
 			[
 				'video_id' => false,
+				'layout'   => 'responsive',
+				'width'    => $this->args['width'],
+				'height'   => $this->args['height'],
+				'fallback' => '',
 			]
 		);
 
 		if ( empty( $args['video_id'] ) ) {
 			return AMP_HTML_Utils::build_tag(
 				'a',
 				[
-					'href'  => esc_url( $args['url'] ),
+					'href'  => esc_url( $url ),
```

Comment:
Just realized that this needs to rather be `esc_url_raw()` since the HTML entity encoding will be handled by the DOM. If there are other places where `esc_url()` is being used in this way, it should be updated to be `esc_url_raw()` as well.